### PR TITLE
Build out the all tags page

### PR DESCRIPTION
**Pull request by briancbarrow**

### CODE REVIEW

File: blog/src/components/authors/Hero.astro [hunk 1]
```diff
@@ -3,7 +3,8 @@ import LeftDecoration from "../../shared/components/decoration/LeftDecoration.as
 import RightDecoration from "../../shared/components/decoration/RightDecoration.astro";
 import Svg from "../../shared/components/general/Svg.astro";
 
-const { title } = Astro.props;
+const { title, svg } = Astro.props;
+// console.log({ svg, title });
```
Comment (lukeocodes) [inline blog/src/components/authors/Hero.astro:7]:
```suggestion
```

File: blog/src/components/tags/Hero.astro [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+import LeftDecoration from "../../shared/components/decoration/LeftDecoration.astro";
+import RightDecoration from "../../shared/components/decoration/RightDecoration.astro";
+import Svg from "../../shared/components/general/Svg.astro";
+
+const { title, svg } = Astro.props;
+// console.log({ svg, title });
```
Comment (lukeocodes) [inline blog/src/components/tags/Hero.astro:7]:
```suggestion
```

File: blog/src/assets/styles/blog.css [hunk 3]
```diff
@@ -7,32 +7,39 @@
 
 .bg-starburst-gradient {
 	background-image: url("/starburst-gradient.svg");
+}
+
+.bg-tag-gradient {
+	background-image: url("/tag-gradient.svg");
+}
+
+.bg-svg {
```
Comment (lukeocodes) [inline blog/src/assets/styles/blog.css:16]:
```suggestion
.bg-responsive-svg {
```

File: blog/src/assets/styles/blog.css [hunk 4]
```diff
@@ -7,32 +7,39 @@
...
 	background-repeat: no-repeat;
 	background-position: center bottom;
 	background-size: 60%;
 }
 
 @media screen(sm) {
-	.bg-starburst-gradient {
+	.bg-svg {
```
Comment (lukeocodes) [inline blog/src/assets/styles/blog.css:23]:
```suggestion
	.bg-responsive-svg {
```

File: blog/src/assets/styles/blog.css [hunk 5]
```diff
@@ -7,32 +7,39 @@
...
 		background-position: center bottom 1rem;
 		background-size: auto;
 	}
 }
 
 @media screen(md) {
-	.bg-starburst-gradient {
+	.bg-svg {
```
Comment (lukeocodes) [inline blog/src/assets/styles/blog.css:30]:
```suggestion
	.bg-responsive-svg {
```

File: blog/src/assets/styles/blog.css [hunk 6]
```diff
@@ -7,32 +7,39 @@
...
 		background-position: center bottom 2rem;
 	}
 }
 
 @media screen(lg) {
-	.bg-starburst-gradient {
+	.bg-svg {
```
Comment (lukeocodes) [inline blog/src/assets/styles/blog.css:36]:
```suggestion
	.bg-responsive-svg {
```

File: blog/src/assets/styles/blog.css [hunk 7]
```diff
@@ -7,32 +7,39 @@
...
 		background-position: center bottom 3rem;
 	}
 }
 
 @media screen(xl) {
-	.bg-starburst-gradient {
+	.bg-svg {
```
Comment (lukeocodes) [inline blog/src/assets/styles/blog.css:42]:
```suggestion
	.bg-responsive-svg {
```

File: blog/src/pages/authors/index.astro [hunk 8]
```diff
@@ -64,7 +64,7 @@ const schema = {
 	<Meta slot="og:title" property="og:title" content="All authors - Deepgram Blog ⚡️" />
 	<Meta slot="og:description" property="og:description" content="All the authors writing content on the Deepgram blog." />
 
-	<ContrastSection contrast="darkCharcoal" class="bg-starburst-gradient" background="white" bottomDivider="eclipse-divider">
+	<ContrastSection contrast="darkCharcoal" class="bg-starburst-gradient bg-svg" background="white" bottomDivider="eclipse-divider">
```
Comment (lukeocodes) [inline blog/src/pages/authors/index.astro:67]:
```suggestion
	<ContrastSection contrast="darkCharcoal" class="bg-starburst-gradient bg-responsive-svg" background="white" bottomDivider="eclipse-divider">
```

File: blog/src/pages/tags/index.astro [hunk 9]
```diff
@@ -2,9 +2,22 @@
...
 import Backlink from "../../components/nav/Backlink.astro";
+
+let tags = [];
+const allPosts = await Astro.glob("../../content/blog/posts/**/index.md");
+
+allPosts.forEach((post, i) => {
+	tags = [...tags, ...post.frontmatter.tags];
+});
+const uniqueTags = Array.from(new Set(tags));
+// console.log({ uniqueTags });
```
Comment (lukeocodes) [inline blog/src/pages/tags/index.astro:20]:
```suggestion
```

File: blog/src/pages/tags/index.astro [hunk 10]
```diff
@@ -2,9 +2,22 @@
...
+import Hero from "../../components/tags/Hero.astro";
+import Link from "../../shared/components/general/Link.astro";
 import Backlink from "../../components/nav/Backlink.astro";
+
+let tags = [];
+const allPosts = await Astro.glob("../../content/blog/posts/**/index.md");
+
+allPosts.forEach((post, i) => {
+	tags = [...tags, ...post.frontmatter.tags];
+});
```
Comment (lukeocodes) [inline blog/src/pages/tags/index.astro:18]:
```suggestion
allPosts.forEach((post) => {
  tags.push(...post.frontmatter.tags)
});

```

File: blog/src/pages/tags/index.astro [hunk 11]
```diff
@@ -2,9 +2,22 @@
...
+const uniqueTags = Array.from(new Set(tags));
```
Comment (lukeocodes) [inline blog/src/pages/tags/index.astro:19]:
```suggestion
tags = [...new Set(tags)];
```

File: blog/src/pages/tags/index.astro [hunk 12]
```diff
@@ -13,17 +26,27 @@ import Backlink from "../../components/nav/Backlink.astro";
...
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
+	<PrimarySection class="my-16 mx-16 ml-20">
+		<div class="flex justify-end mb-5 ">
+			<Link href="/posts" class={`text-lg font-semibold stroke-[3]`} icon="arrow-right" iconSuffix><span>View All Posts</span></Link>
+		</div>
+		<ul class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 gap-y-8 lg:grid-cols-4 auto-cols-max grid-flow-row-dense">
+			{uniqueTags.map((tag) => (
```
Comment (lukeocodes) [inline blog/src/pages/tags/index.astro:42]:
```suggestion
			{tags.map((tag) => (
```

**Review by lukeocodes — APPROVED** [review]
Added some code suggestions. Main one is removing those commented out console.logs though lol

File: blog/src/pages/tags/index.astro [hunk 13]
```diff
@@ -13,17 +26,27 @@ import Backlink from "../../components/nav/Backlink.astro";
...
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
+	<PrimarySection class="my-16 mx-16 ml-20">
+		<div class="flex justify-end mb-5 ">
+			<Link href="/posts" class={`text-lg font-semibold stroke-[3]`} icon="arrow-right" iconSuffix><span>View All Posts</span></Link>
+		</div>
```
Comment (lukeocodes) [inline blog/src/pages/tags/index.astro:40]: <LinkCardListTitle> will do what you need here, won't it?

File: blog/src/pages/tags/index.astro [hunk 14]
```diff
@@ -13,17 +26,27 @@ import Backlink from "../../components/nav/Backlink.astro";
...
+	</ContrastSection>
 
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
-		<div>Tag goes here</div>
+	<PrimarySection class="my-16 mx-16 ml-20">
```
Comment (lukeocodes) [inline blog/src/pages/tags/index.astro:37]:
```suggestion
	<PrimarySection class="my-16 mx-16 ml-20">
```

This is what is breaking the responsiveness. PrimarySection is a class that has concrete margins to make sure every part of the content lines up vertically. What is it you're trying to achieve with this horizonal margin?

**Review by lukeocodes — CHANGES_REQUESTED** [review]

**Comment by lukeocodes** [comment]
image

File: blog/src/pages/tags/index.astro (hunk 13 above)
Comment (briancbarrow) [reply blog/src/pages/tags/index.astro:40]: I initially tried it but thought it needed a title so I went with something else. I see now that it doesn't need that so I switched it back.

File: blog/src/pages/tags/index.astro [hunk 15]
```diff
@@ -2,9 +2,23 @@
...
+const allPosts = await Astro.glob("../../content/blog/posts/**/index.md");
+
+allPosts.forEach((post) => {
+	tags.push(...post.frontmatter.tags);
+});
+
+tags = [...new Set(tags)];
 ---
 
 <Layout>
```
Comment (lukeocodes) [inline blog/src/pages/tags/index.astro:24]: Oops i forgot to mention structured data. Please see /pages/tags/[tag]/[...page].astro for the schema.

Or something like this?

```suggestion
// build json schema object
const schema = {
	"@context": "https://schema.org",
	"@type": "BreadcrumbList",
	itemListElement: [
		{
			"@type": "ListItem",
			position: 1,
			name: "Deepgram Home",
			item: "https://deepgram.com",
		},
		{
			"@type": "ListItem",
			position: 2,
			name: "Blog",
			item: Astro.url.origin,
		},
		{
			"@type": "ListItem",
			position: 3,
			name: "Tags",
		},
	],
};
---

<Layout>
	<JsonLD slot="json:ld" {schema} />
```

**Review by lukeocodes — CHANGES_REQUESTED** [review]

**Review by lukeocodes — APPROVED** [review]

**Pull request merged**